src/python/pants/backend/codegen/protobuf/buf.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
```

Comment:
> I put the external tool for Buf in backend/codegen/protobuf instead of in backend/codegen/protobuf/lint/buf since I imagine it can and will be reused to future Buf functionality (like the BC breaking check).

Imo it's better to put this in `backend/codegen/lint/buf` for now. We can always move the code later if we do generalize. Pants has so many moving parts that we try to avoid "premature generalization"